_posts/2020-03-06-introducing-mirrormaker2.md

```diff
@@ -0,0 +1,247 @@
+as we're using a Kafka Connect framework to make our connection.
+
+The `clusters` properties define the Kafka clusters being synchronized, and the `mirrors` properties define the MirrorMaker 2.0 connectors.
+
+You can build a more elaborate configuration.
+For example, you can add configuration to include replica nodes, which is the number of replica nodes in the Kafka Connect group, and TLS or SASL authentication for the source and target cluster too.
+Typical and sensible choices.
+
+Here we create 3 replica nodes and use TLS authentication.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1alpha1
+kind: KafkaMirrorMaker2
+metadata:
+spec:
+  version: {DefaultKafkaVersion}
+  replicas: 3
+  connectCluster: "my-cluster-target"
+    clusters:
+    - alias: "my-cluster-source"
+    authentication:
+      certificateAndKey:
+        certificate: source.crt
+        key: source.key
+        secretName: my-user-source
+      type: tls
+    bootstrapServers: my-cluster-source-kafka-bootstrap:9092
+    tls:
+      trustedCertificates:
+      - certificate: ca.crt
+        secretName: my-cluster-source-cluster-ca-cert
+  - alias: "my-cluster-target"
+    authentication: <10>
+      certificateAndKey:
+        certificate: target.crt
+        key: target.key
+        secretName: my-user-target
+      type: tls
+    bootstrapServers: my-cluster-target-kafka-bootstrap:9092
+    config:
+      config.storage.replication.factor: 1
+      offset.storage.replication.factor: 1
+      status.storage.replication.factor: 1
+    tls:
+      trustedCertificates:
+      - certificate: ca.crt
+        secretName: my-cluster-target-cluster-ca-cert
+```
+
+People can be confused by the idea of _replicas_ and _replication_ through MirrorMaker,
+that's why you'll see replication between clusters often referred to as _mirroring_.
+Just thought I'd bring that up in case I've been inconsistent in this post.
+
+If you don't want to leave the defaults, you can also include configuration for the **_MirrorMaker 2.0 connectors_** and related **_internal topics_**.
+
+The `config` overrides the default configuration options, so here we alter the replication factors for the **_internal topics_**:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1alpha1
+kind: KafkaMirrorMaker2
+metadata:
+  spec:
+  # ...
+    mirrors:
+    # ...
+    sourceConnector:
+      config:
+        replication.factor: 1
+        offset-syncs.topic.replication.factor: 1
+    heartbeatConnector:
+      config:
+        heartbeats.topic.replication.factor: 1
+    checkpointConnector:
+      config:
+        checkpoints.topic.replication.factor: 1
+```
+
+You can see the full `spec` options in the [KafkaMirrorMaker2 schema reference](https://strimzi.io/docs/master/#type-KafkaMirrorMaker2-reference).
+
+If, at this point, you're wondering what happens if you're using the old version of MirrorMaker.
+It's still supported, but you will need to updated to the format supported by MirrorMaker 2.0.
```

Comment:
Typo: `update` 
This line is a bit unclear - what do you mean by 'update to the format supported by MM2'? There are plans in Kafka to have a 'legacy' mode in MirrorMaker 2.0 (but it's not in Kafka 2.4.0) that will create topics without the cluster prefix and not do the topic config mirroring (which I think are the main behavioural changes between MM1 and 2).

Reply:
Changed to:

> If, at this point, you're wondering what happens if you're using the old version of MirrorMaker,
> it's still supported.
> And there are also plans to have a _legacy mode_ in MirrorMaker 2.0 that creates topics without the cluster prefix, 
> and doesn't do the topic configuration mirroring.
> Basically, turning off the main differences between the original and the new version of MirrorMaker.